lab3/client/src/reducer.js

```diff
@@ -0,0 +1,42 @@
+import constants from './constants';
+import Generator from './models/generator';
+
+export default function reducer (state, action) {
+	switch (action.type) {
+		case constants.actions.EXAMPLE:
+			state.example = action.payload;
+			return state;
+		case constants.actions.INCREMENT:
+			state.counter += action.payload;
+			return state;
+		break;
+		case constants.actions.BUY_GENERATOR:
+			const currentGenerator = action.generatorClicked || 0;
```

Comment:
This will not pass the unit test for the `reducer.test.js` considering the test is using different type of _payload_. You should read a bit on what payload it is in the tests.

Reply:
still not passing the unit test.

Reply:
@rcliao I know that i diverted it because i wanted to send particular generator each time. But when i do "npm run test" or i upload the code on github it still passes the test. Why so then?